DuoUniversal/Client.cs

```diff
@@ -148,6 +158,47 @@ public async Task<IdToken> ExchangeAuthorizationCodeFor2faResult(string duoCode,
         }
 
 
+        /// <summary>
+        /// Send the authorization code provided by Duo back to Duo in exchange for an Id Token authenticating the user and
+        /// providing details about the authentication.
+        /// Will raise a DuoException if the username does not match the Id Token.
+        /// </summary>
+        /// <param name="duoCode">The one-time use code issued by Duo</param>
+        /// <param name="username">The username expected to have authenticated with Duo</param>
+        /// <returns>An IdToken authenticating the user and describing the authentication</returns>
+        public async Task<IdToken> ExchangeAuthorizationCodeFor2faResult(string duoCode, string username)
+        {
+            TokenResponse tokenResponse = await ExchangeAuthorizationCodeResponse(duoCode);
+            return IdTokenFromResponse(tokenResponse, username);
+
+        }
+
+        /// <summary>
+        /// Send the authorization code provided by Duo back to Duo in exchange for an SAML response, used for some integrations.
+        /// Will raise a DuoException if the username does not match the Id Token.
+        /// </summary>
+        /// <param name="duoCode">The one-time use code issued by Duo</param>
+        /// <param name="username">The username expected to have authenticated with Duo</param>
+        /// <returns>A string authenticating the user and containing the saml response</returns>
+        public async Task<string> ExchangeAuthorizationCodeForSamlResponse(string duoCode, string username)
+        {
+            string samlResponse;
+            TokenResponse tokenResponse = await ExchangeAuthorizationCodeResponse(duoCode);
+            //checking if the IdToken valid before assigning saml response
+            try
+            {
+                IdTokenFromResponse(tokenResponse, username);

```

Comment:
The reason I want to rename the method is so that it does not look out of place here.  Someone in the future might look here and not know why we are calling this method when we don't use the return value.  Calling out that it validates the token should make it more obvious why it's necessary.

Reply:
My way of thinking is, as this method returns the IdToken it should be named like this. I didn't see a good way to split 2 methods - one for returning the token, another for it's validation as they would look almost identical. But let me check tomorrow again, maybe a week off will give a fresh look :)